waspc/src/Wasp/Generator/Job/Process.hs

```diff
@@ -110,40 +112,51 @@ runNodeCommandAsJob fromDir command args jobType chan = do
-          `catchIOError` ( \e ->
-                             if isDoesNotExistError e
-                               then return (ExitFailure 1, "", "Command 'node' not found.")
-                               else ioError e
-                         )
-      return $ case exitCode of
-        ExitFailure _ ->
-          Left
-            ( "Running 'node --version' failed: " ++ stderr
-                ++ " "
-                ++ waspNodeRequirementMessage
-            )
-        ExitSuccess -> case parseNodeVersion stdout of
-          Nothing ->
-            Left
-              ( "Wasp failed to parse node version."
-                  ++ " This is most likely a bug in Wasp, please file an issue."
-              )
-          Just version -> Right version
+getNodeVersion :: IO (Either String SemanticVersion)
+getNodeVersion = do
+  (exitCode, stdout, stderr) <-
+    P.readProcessWithExitCode "node" ["--version"] ""
+      `catchIOError` ( \e ->
+                         if isDoesNotExistError e
+                           then return (ExitFailure 1, "", "Command 'node' not found.")
+                           else ioError e
+                     )
+  return $ case exitCode of
+    ExitFailure _ ->
+      Left
+        ( "Running 'node --version' failed: " ++ stderr
+            ++ " "
+            ++ waspNodeRequirementMessage
+        )
+    ExitSuccess -> case parseNodeVersion stdout of
+      Nothing ->
+        Left
+          ( "Wasp failed to parse node version."
+              ++ " This is most likely a bug in Wasp, please file an issue."
+          )
+      Just version -> Right version
+
+parseNodeVersion :: String -> Maybe SemanticVersion
+parseNodeVersion nodeVersionStr =
+  case nodeVersionStr R.=~ ("v([^\\.]+).([^\\.]+).(.+)" :: String) of
+    ((_, _, _, [majorStr, minorStr, patchStr]) :: (String, String, String, [String])) -> do
+      mjr <- readMaybe majorStr
+      mnr <- readMaybe minorStr
+      ptc <- readMaybe patchStr
+      return $ SemanticVersion mjr mnr ptc
+    _ -> Nothing
 
-    parseNodeVersion :: String -> Maybe (Int, Int, Int)
-    parseNodeVersion nodeVersionStr =
-      case nodeVersionStr R.=~ ("v([^\\.]+).([^\\.]+).(.+)" :: String) of
-        ((_, _, _, [majorStr, minorStr, patchStr]) :: (String, String, String, [String])) -> do
-          major <- readMaybe majorStr
-          minor <- readMaybe minorStr
-          patch <- readMaybe patchStr
-          return (major, minor, patch)
-        _ -> Nothing
+makeNodeVersionMismatchMessage :: SemanticVersion -> String
+makeNodeVersionMismatchMessage nodeVersion =
+  unwords
+    [ "Your node version does not match Wasp's requirements.",
+      "You are running node" ++ show nodeVersion ++ ".",
+      waspNodeRequirementMessage
+    ]
 
-    waspNodeRequirementMessage =
-      "Wasp requires node >= " ++ C.nodeVersionAsText ++ " ."
-        ++ " Check Wasp docs for more details: https://wasp-lang.dev/docs#requirements ."
+waspNodeRequirementMessage :: String
```

Comment:
Ha you probably could have let this be part of `makeNodeVersionMismatchMessage` :D. But doesn't really make a difference, all good.